# --- ROCK-COPYRIGHT-NOTE-BEGIN ---
# 
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# Please add additional copyright information _after_ the line containing
# the ROCK-COPYRIGHT-NOTE-END tag. Otherwise it might get removed by
# the ./scripts/Create-CopyPatch script. Do not edit this copyright text!
# 
# ROCK Linux: rock-src/package/base/dietlibc/pkg_patch/pkg_linux24.patch
# ROCK Linux is Copyright (C) 1998 - 2004 Clifford Wolf
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# 
# --- ROCK-COPYRIGHT-NOTE-END ---

--- linux-2.4.22/scripts/mkdep.c.orig	2003-09-20 20:54:57.000000000 +0200
+++ linux-2.4.22/scripts/mkdep.c	2003-09-20 20:55:43.000000000 +0200
@@ -38,7 +38,9 @@
 #include <string.h>
 #include <unistd.h>
 
+#ifndef __dietlibc__
 #include <sys/fcntl.h>
+#endif
 #include <sys/mman.h>
 #include <sys/stat.h>
 #include <sys/types.h>
--- linux-2.4.22/lib/gen_crc32table.c.orig	2003-09-20 21:51:19.000000000 +0200
+++ linux-2.4.22/lib/gen_crc32table.c	2003-09-20 21:51:48.000000000 +0200
@@ -1,3 +1,6 @@
+#ifdef __dietlibc__
+#define _GNU_SOURCE
+#endif
 #include <stdio.h>
 #include "crc32defs.h"
 #include <sys/types.h>
